### PR TITLE
db-immutaliser

### PR DESCRIPTION
**Pull request by amesgen**
This PR adds a new small developer tool, db-immutaliser, which can be used to copy a chain from a volatile DB to an immutable DB.

 - Tools like immdb-server or the external [cardano-streamer](https://github.com/lehins/cardano-streamer) by @lehins currently only work with the immutable part of a ChainDB. Instead of adding support to all of these tools to also work with a volatile DB, db-immutaliser can solve that task in a composable manner.

   (Context: This PR was motivated by @lehins tracking down a Ledger bug at ZuriHac using his cardano-streamer, where the block exhibiting the bug turned out to reside in the VolatileDB.)

 - Our other prominent tool, db-analyser, has support for running with a full ChainDB (including a VolatileDB), but it is basically never used (in particular, it does not support `--analyse-from`), and it unconditionally requires loading a ledger snapshot for initial chain selection[^1]. In contrast, db-immutaliser works without a ledger snapshot ATM, using the best volatile chain without performing any validation. It could be extended if validating the volatile chain is crucial.

   Therefore, this PR also removes support for using the full ChainDB, making `--only-immutable-db` the default.

[^1]: Currently, using db-analyser with the ImmutableDB *also* always requires a ledger snapshot, but at least it only requires one for the given slot specified by `--analyse-from`, and this can also be fixed much more simply than for the full ChainDB case.

### CODE REVIEW

**Review by dnadales — APPROVED** [review]
Thanks. 

We might want to check the "Unchanged files with check annotations" report as well.

File: ouroboros-consensus-cardano/changelog.d/20240610_154105_alexander.esgen_db_immutaliser.md
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus-cardano/changelog.d/20240610_154105_alexander.esgen_db_immutaliser.md:1]: My recollection is that we don't track changes in unstable sublibs or in aux tools

File: ouroboros-consensus-cardano/app/db-immutaliser.hs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE ApplicativeDo  #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Main (main) where
+
+import           Cardano.Crypto.Init (cryptoInit)
+import           Cardano.Tools.DBImmutaliser.Run
```
[the remark this answers is not in the document]
Comment (amesgen) [reply ouroboros-consensus-cardano/app/db-immutaliser.hs:7]: I have no strong opinion, I chose this capitalization as db-analyser and db-truncater use the same one.

File: ouroboros-consensus-cardano/app/db-immutaliser.hs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import           Main.Utf8 (withStdTerminalHandles)
+import           Options.Applicative
+main :: IO ()
+main = withStdTerminalHandles $ do
+    cryptoInit
+    run =<< execParser optsParser
```
[the remark this answers is not in the document]
Comment (amesgen) [reply ouroboros-consensus-cardano/app/db-immutaliser.hs:13]: Will change :+1:

**Pull request merged**